### PR TITLE
Blog post "Generating D3 code with LLMs" (new author rlewis)

### PR DESCRIPTION
**Pull request by rlewis-sl**
Please add a direct link to your post here:

https://rlewis-sl.github.io/blog/2024/03/26/generating-d3-code-with-llms.html

Have you (please tick each box to show completion):

 - [x] Added your blog post to a single category?
 - [x] Added a brief summary for your post? Summaries should be roughly two sentences in length and give potential readers a good idea of the contents of your post.
 - [x] Checked that the build passes?
 - [x] Checked your spelling (you can use `npm spellcheck` if that's your thing)
 - [x] Ensured that your author profile contains a profile image, and a brief description of yourself? (make it more interesting than just your job title!)
 - [x] Optimised any images in your post? They should be less than 100KBytes as a general guide.

Posts are reviewed / approved by your Regional Tech Lead.

### CODE REVIEW

File: _posts/2024-03-26-generating-d3-code-with-llms.markdown [hunk 1]
```diff
@@ -0,0 +1,420 @@
+---
+title: 'Generating D3 code with LLMs'
+author: rlewis
+date: 2024-03-26 00:00:00 Z
+summary: Can AI write software? My team at Scott Logic wanted to find out whether LLMs could generate JavaScript code to draw a D3 chart in a web page. Read about how it went in this blog post.
+categories:
+- Artificial Intelligence
+---
+
+My team at Scott Logic wanted to see how well large language models (LLMs) could perform on a fully automated coding task. We decided to try using an LLM to generate JavaScript code to display a chart in a web page using the [D3 library](https://d3js.org/).
```
Comment (jmoore-scottlogic) [inline _posts/2024-03-26-generating-d3-code-with-llms.markdown]: "to try using" -> "to use"

File: _posts/2024-03-26-generating-d3-code-with-llms.markdown [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+
+We knew there would be many examples of D3 code available on the internet and so LLMs were likely to have had large amounts of relevant training data. But the task was also more complex than the short, well-defined and self-contained coding problems that LLMs are normally evaluated against. None of us had much experience with LLMs, but through experimentation we soon learned more about what to expect. We found that LLMs could often succeed in the task, but only if we provided them with the right kind of help.
```
Comment (jmoore-scottlogic) [inline _posts/2024-03-26-generating-d3-code-with-llms.markdown]: "were likely to have had large" -> "were likely to have access to large"

Comment (jmoore-scottlogic) [inline _posts/2024-03-26-generating-d3-code-with-llms.markdown]: "But the task was also more complex" -> "However, the task we presented was more complex"

Comment (jmoore-scottlogic) [inline _posts/2024-03-26-generating-d3-code-with-llms.markdown]: "None of us had much experience" -> "No one on the team had much experience"

File: _posts/2024-03-26-generating-d3-code-with-llms.markdown [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+title: 'Generating D3 code with LLMs'
+author: rlewis
+date: 2024-03-26 00:00:00 Z
+summary: Can AI write software? My team at Scott Logic wanted to find out whether LLMs could generate JavaScript code to draw a D3 chart in a web page. Read about how it went in this blog post.
+categories:
+- Artificial Intelligence
+---
+
+My team at Scott Logic wanted to see how well large language models (LLMs) could perform on a fully automated coding task. We decided to try using an LLM to generate JavaScript code to display a chart in a web page using the [D3 library](https://d3js.org/).
+
```
Comment (jmoore-scottlogic) [inline _posts/2024-03-26-generating-d3-code-with-llms.markdown:11]: ", but through experimentation" -> ", but through experimentation,"

**Review by jmoore-scottlogic — COMMENTED** [review]
Overall looks very good. A few minor grammar issues, might be worth just running it through an LLM to check the rest.

File: _posts/2024-03-26-generating-d3-code-with-llms.markdown (hunk 1 above)
Comment (rlewis-sl) [reply _posts/2024-03-26-generating-d3-code-with-llms.markdown]: I think "to try using" captures my meaning.

File: _posts/2024-03-26-generating-d3-code-with-llms.markdown (hunk 3 above)
Comment (rlewis-sl) [reply _posts/2024-03-26-generating-d3-code-with-llms.markdown:11]: Are you suggesting a comma after "experimentation"? I wouldn't put a comma there.

File: _posts/2024-03-26-generating-d3-code-with-llms.markdown (hunk 2 above)
Comment (rlewis-sl) [reply _posts/2024-03-26-generating-d3-code-with-llms.markdown]: I made a bigger change. Does that seem better?

Comment (rlewis-sl) [reply _posts/2024-03-26-generating-d3-code-with-llms.markdown]: Sure. Why not?

Comment (rlewis-sl) [reply _posts/2024-03-26-generating-d3-code-with-llms.markdown]: I changed to "the task we had in mind was more complex"

**Pull request merged**